src/spiral_stc.cpp

```diff
@@ -90,64 +101,193 @@ namespace full_coverage_path_planner
 
     return global_path;
   }
+
   std::list<gridNode_t> SpiralSTC::spiral(std::vector<std::vector<bool>> const &grid, std::list<gridNode_t> &init,
-                                          std::vector<std::vector<bool>> &visited)
+                                          double &yaw_start, std::vector<std::vector<bool>> &visited)
   {
-    int nRows = grid.size();
-    int nCols = grid[0].size();
-    // Spiral filling of the open space
-    // Copy incoming list to 'end'
-    std::list<gridNode_t> pathNodes(init);
-    // Create iterator for gridNode_t list and let it point to the last element of end
-    std::list<gridNode_t>::iterator it = --(pathNodes.end());
-    if (pathNodes.size() > 1) // if list is length 1, keep iterator at end
-      it--;                   // Let iterator point to second to last element
-
+    std::list<gridNode_t> pathNodes(init); // Copy incoming list to pathNodes
+    std::list<gridNode_t>::iterator it = --(pathNodes.end()); // Create iterator and let it point to the last element of end
+    if (pathNodes.size() > 1) // If list is length 1, keep iterator at end
+    {
+      it--; // Let iterator point to second to last element
+    }
     gridNode_t prev = *(it);
+
+    // Initialize spiral direction towards robot's y-axis
+    int dx = 0;
+    int dy = 1;
+    int dx_prev;
+    double yaw1 = yaw_start; // This value will change later on
+
+    // Mark initial footprint as visited
+    std::vector<nav2_costmap_2d::MapLocation> init_cells;
+    if (!computeFootprintCells(pathNodes.back().pos.x, pathNodes.back().pos.y, yaw1, "tool", init_cells))
+    {
+      RCLCPP_ERROR(rclcpp::get_logger("FullCoveragePathPlanner"), "Starting footprint seems to be out of bounds!");
+    }
+    for (auto const cell : init_cells)
+    {
+      visited[cell.y][cell.x] = eNodeVisited;
+    }
+
+    std::vector<nav2_costmap_2d::MapLocation> man_grids;
     bool done = false;
     while (!done)
     {
-      // Initialize spiral direction towards y-axis
-      int dx = 0;
-      int dy = 1;
-      int dx_prev;
       if (it != pathNodes.begin())
       {
-        // turn ccw
+        // Turn counter-clockwise
         dx = pathNodes.back().pos.x - prev.pos.x;
         dy = pathNodes.back().pos.y - prev.pos.y;
+        yaw1 = std::atan2(dy,dx); // Keep overwriting the orientation according to the last two nodes
         dx_prev = dx;
         dx = -dy;
         dy = dx_prev;
       }
-      // This condition might change in the loop bellow
-      done = true;
-      // loop over the four possible directions: up, right, down, left
-      for (size_t i = 0; i < 4; ++i)
+      done = true; // This condition might change in the loop below
+      // Loop over the three possible directions: left, forward, right (directions taken before counter-clockwise turn)
+      for (size_t i = 0; i < 3; ++i)
       {
+        int x1 = pathNodes.back().pos.x;
```

Comment:
Also in these variables please use more descriptive names ( some of them were already there )

Reply:
I find 'd' for distance descriptive..

Reply:
See the latest commit.